src/wavesurfer.js

```diff
@@ -84,6 +85,23 @@ var WaveSurfer = {
         this.createPeakCache();
 
         this.isDestroyed = false;
+
+        // Responsive waveform on resize and device orientation change
+        // the waveform will be responsive by default
+        // if the responsive parameter is not set or set to true
+        // setting the responsive paramater to false will disable the responsive waveform
+        var prevWidth = 0;
+        this.onResize = WaveSurfer.util.debounce(function () {
+            if (prevWidth != this.wrapper.clientWidth) {
+                prevWidth = this.wrapper.clientWidth;
+                this.render();
+                this.progress(WaveSurfer.backend.getPlayedPercents());
```

Comment:
I could not find the definition of neither `render` nor `progress`. I think calling `drawBuffer` should be enough once the redrawing bug discussed in #1127 is resolved.